.gitlab-ci.yml

```diff
@@ -0,0 +1,162 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# GitLab CI/CD for Python
+
+workflow:
+  rules:
+    - if: $CI_EXTERNAL_PULL_REQUEST_IID &&
+          $CI_EXTERNAL_PULL_REQUEST_SOURCE_REPOSITORY == $CI_EXTERNAL_PULL_REQUEST_TARGET_REPOSITORY &&
+          ( $CI_EXTERNAL_PULL_REQUEST_TARGET_BRANCH_NAME == $CI_DEFAULT_BRANCH ||
+            $CI_EXTERNAL_PULL_REQUEST_TARGET_BRANCH_NAME =~ /hackathon\/.+/ )
+    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
+
+default:
+  tags:
+    - docker
+
+variables:
+  PYTHON_IMAGE: python:3.9
+  RUN_DIR: ./cicd/runtime
+  # Configure mysql environment variables (https://hub.docker.com/_/mysql/)
+  MYSQL_ROOT_PASSWORD: root
+  MYSQL_USER: runner
+  MYSQL_PASSWORD: mysql
+
+.python:
+  stage: test
+  image: $PYTHON_IMAGE
+  services:
+  - mysql:5.7
+  only:
+    changes:
+      - .gitlab-ci.yml
+      - pyproject.toml
+      - "*.py"
+      - src/python/**/*.py
+
+## Prepare and load the Python virtual environment
+
+python:prepare_venv:
+  extends: .python
+  artifacts:
+    paths:
+      - $RUN_DIR
+  before_script:
+    - python --version  # for debugging
+    - python -m venv $RUN_DIR/venv
+    - source $RUN_DIR/venv/bin/activate
+  script:
+    - pip install -e .[cicd]
```

Comment:
We might want to add the cicd dependencies in pyproject.toml file

Reply:
I completely forgot about that :facepalm: